### PR TITLE
doc: Add Dev Containers related documentation.

### PR DESCRIPTION
**Pull request by huxuan**
----
📚 Documentation preview 📚: https://ss-python--305.org.readthedocs.build/en/305/

### CODE REVIEW

File: docs/dev/dev-env.md
Comment (msclock) [inline docs/dev/dev-env.md:1]: this page shows

File: docs/dev/dev-env.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Environment Management
+# Development Environment
+
+This page show the approach to manage development environment. To simplify the process, a unified `Makefile` is maintained at the root directory of the repo. In other words, all the `make` related commands are supposed to run there.
```
Comment (msclock) [inline docs/dev/dev-env.md:3]: This page shows the ...

File: docs/dev/proj.md [hunk 2]
```diff
@@ -50,6 +50,34 @@ pipx install copier
...
+
+    ```{image} /_static/images/bootstrap-dev-container-github.png
+    :alt: Bootstrap Dev Container on GitHub.
+    :target: https://github.com/serious-scaffold/ss-python/actions/workflows/devcontainer.yml
+    ```
+
+    :::
+
+    :::{tab-item} GitLab
+    :sync: gitlab
```
Comment (msclock) [inline docs/dev/proj.md:69]: maybe pipelines

File: docs/dev/proj.md [hunk 3]
```diff
@@ -50,6 +50,34 @@ pipx install copier
...
+    Open the [Run Pipelines](https://gitlab.com/serious-scaffold/ss-python/-/pipelines/new) page and run pipeline for the default branch.
+
```
Comment (msclock) [inline docs/dev/proj.md:71]: maybe pipelines😅

**Review by msclock — COMMENTED** [review]
👍 everything is good but some typos

File: docs/dev/dev-env.md (hunk 1 above)
Comment (huxuan) [reply docs/dev/dev-env.md]: Done. Thanks for pointing out.

File: docs/dev/proj.md (hunk 3 above)
Comment (huxuan) [reply docs/dev/proj.md:71]: Done, according to the screenshot, I changed the first appearance to "Run pipeline". Let me know if there is any misunderstanding.

File: docs/dev/dev-env.md [hunk 4]
```diff
@@ -15,47 +17,46 @@ pipx install pdm
...
+Deep cleanup will remove the pre-commit hook and the virtual environment alongside the common intermediate files.
 
 ```bash
 make deepclean
 ```
 
 ### Complete cleanup
 
-With this command, the repo will be as if it has been re-cloned. It is useful if we want to start almost from scratch.
+Complete cleanup will make the repo as if it has been re-cloned. It is useful if we want to start from scratch.
```
Comment (msclock) [inline docs/dev/dev-env.md:59]: 'make the repo'  maybe have made a mistakes with the expressing senses of remove or remake.

Comment (huxuan) [reply docs/dev/dev-env.md]: Done, asked ChatGPT with minor refinements. :-)

**Review by msclock — APPROVED** [review]

**Pull request merged**